### DISCUSSION TITLE
Config files for Minestom extensions

### DISCUSSION THREAD

**Discussion by LoisDuplain** [Minestom/Minestom#1450] [Help] [answered]
Hi,

There is nothing in Minestom to handle config.yml config file in Minestom extensions ?

**Comment by KrystilizeNevaDies** [comment, marked as answer]
Nope, you should add one yourself. There are libraries for this such as Sponge's Configurate.

**Reply by LoisDuplain** [reply]
Why not implement it in Minestom? This would avoid redundancy

**Reply by ZakShearman** [reply]
Many projects avoid it as people have their own preferences. I think there are a lot of Minestom users who just wouldn't use a YAML addon. Many use JSON, I use HOCON or straight from env vars myself.

**Reply by LoisDuplain** [reply]
Most of the time I don't use YAML but here I have to port an existing plugin to Minestom
But Sponge Configurate supports JSON, HOCON, YAML and XML so it can be interesting to implement it by default

**Reply by Timofffee** [reply]
>  it can be interesting to implement it by default

Just yesterday I was thinking about adding some kind of config on top of Minestom, but in the end I refused it.
"OK, I'm using this config like this. But what prevents others from using it like this?"
That is, literally! I can use one large config, while someone else can use several small ones, or even use something like duplicate configs (for example, when instances have the same configs with different parameters inside).
It is for this reason that there is no need as such for Minestom to have a default config. It is better to have one more article on the wiki that would explain this and show one of the options.
